### PR TITLE
refactor: support caching refresh_token in localStorage

### PR DESCRIPTION
**Pull request by samblacklock**
## Type

* ### Refactor
  A code change that neither fixes a bug nor adds a feature

## Description

Extension of work already done to expose admin endpoints - adds support for `refresh_token`. We may not need this long term but the current dashboard does make use of it, so adding for completeness.

### CODE REVIEW

File: src/factories/request.js [hunk 1]
```diff
@@ -73,12 +57,14 @@ class RequestFactory {
       : createAuthRequest(config)
 
     promise
-      .then(response => {
-        const credentials = new Credentials(
```
Comment (shaunmaharaj) [inline src/factories/request.js:77]: Thanks for the cleanup Sam! I didn't realize this was the only usage of Credentials() too.

**Review by shaunmaharaj — APPROVED** [review]

File: src/factories/request.js (hunk 1 above)
Comment (samblacklock) [reply src/factories/request.js:77]: @shaunmaharaj seems a bit verbose so I stripped it out, unless you have any objections

Comment (shaunmaharaj) [reply src/factories/request.js:77]: @samblacklock nope, I agree. Go for it!

**Pull request merged**